### PR TITLE
Update Role Button Example

### PR DESCRIPTION
**Pull request by SpeedLight-Dev**
The changes include better code explanation and the code has been rendered to be more generalized.

### CODE REVIEW

**Review by Lulalaby — APPROVED** [review]

**Review by Tommins-IT — APPROVED** [review]

**Comment by Lulalaby** [comment]
Fix the codespell error please @SpeedLight-Dev

**Comment by SpeedLight-Dev** [comment]
> Fix the codespell error please @SpeedLight-Dev

Done!

**Review by Lulalaby — APPROVED** [review]

**Review by ChickenDevs — CHANGES_REQUESTED** [review]
Anywhere you removed id(s), we should put in `...` to show the end user where to put their ids.

**Comment by SpeedLight-Dev** [comment]
> Anywhere you removed id(s), we should put in `...` to show the end user where to put their ids.

Ok, I'll do it right now! :)

**Review by Lulalaby — APPROVED** [review]

File: examples/views/button_roles.py [hunk 1]
```diff
@@ -11,11 +11,8 @@
 Make sure to load this cog when your bot starts!
 """
 
-# this is the list of role IDs that will be added as buttons.
-role_ids = [
-    846383888003760217, 846383888036134932
-]
-
+# This list will contain role IDs that will be added as buttons split by a comma ",".
```
Comment (BobDotCom) [inline examples/views/button_roles.py:14]:
```suggestion
# This list will contain role IDs that will be added as buttons.
```
Shouldn't need to specify the comma part.

File: examples/views/button_roles.py [hunk 2]
```diff
@@ -26,11 +23,12 @@ def __init__(self, role: discord.Role):
                          style=discord.enums.ButtonStyle.primary, custom_id=str(role.id))
 
     async def callback(self, interaction: discord.Interaction):
-        """This function will be called any time a user clicks on this button
+        """
+        This function will be called any time a user clicks on this button
```
Comment (BobDotCom) [inline examples/views/button_roles.py:27]:
```suggestion
        """This function will be called any time a user clicks on this button
```
consistency with project code style

File: examples/views/button_roles.py [hunk 3]
```diff
@@ -26,11 +23,12 @@ def __init__(self, role: discord.Role):
...
 
         Parameters
         ----------
-        interaction : discord.Interaction
+        interaction: discord.Interaction
             The interaction object that was created when the user clicked on the button
```
Comment (BobDotCom) [inline examples/views/button_roles.py:32]: Could we just remove this from the docstring? Doesn't feel like we need docs for examples.

File: examples/views/button_roles.py [hunk 4]
```diff
@@ -57,17 +55,19 @@ async def callback(self, interaction: discord.Interaction):
...
     def __init__(self, bot):
         self.bot = bot
 
     # make sure to set the guild ID here to whatever server you want the buttons in
-    @slash_command(guild_ids=[846383887973482516], description="Post the button role message")
+    @slash_command(guild_ids=[...], description="Post the button role message")
     async def post(self, ctx: commands.Context):
-        """Slash command to post a new view with a button for each role
+        """
+        Slash command to post a new view with a button for each role
```
Comment (BobDotCom) [inline examples/views/button_roles.py:70]:
```suggestion
        """Slash command to post a new view with a button for each role
```
Consistency with project code style.

File: examples/views/button_roles.py [hunk 5]
```diff
@@ -83,15 +83,16 @@ async def post(self, ctx: commands.Context):
 
     @commands.Cog.listener()
     async def on_ready(self):
-        """This function is called every time the bot restarts.
+        """
+        This function is called every time the bot restarts.
```
Comment (BobDotCom) [inline examples/views/button_roles.py:87]:
```suggestion
        """This function is called every time the bot restarts.
```
Consistency with project code style.

**Review by BobDotCom — CHANGES_REQUESTED** [review]

**Comment by SpeedLight-Dev** [comment]
I should have done everything you requested.

**Comment by SpeedLight-Dev** [comment]
Gonna open a new one better.

**Pull request closed without merging**